### PR TITLE
TELCODOCS-1059: Logging & Troubleshooting

### PR DESCRIPTION
**Pull request by StephenJamesSmith**
Create KMMO doc: Logging & Troubleshooting

Jira: https://issues.redhat.com/browse/TELCODOCS-1059

Version: openshift-4.12.z

Preview: https://58761--docspreview.netlify.app/openshift-enterprise/latest/hardware_enablement/kmm-kernel-module-management.html#kmm-troubleshooting_kernel-module-management-operator

Dev:[ @qbarrand](https://github.com/qbarrand)
@chr15p
T&PS:[ @slovern](https://github.com/slovern)
QE:[ @cdvultur](https://github.com/cdvultur)

### CODE REVIEW

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
++
+[source,terminal]
+----
+$ oc logs -fn openshift-kmm-hub deployments/kmm-operator-hub-controller-manager
+----
++
+.Example output
+[source,terminal]
+----
+
```
Comment (StephenJamesSmith) [inline modules/kmm-gathering-data-for-kmm-hub.adoc:38]: @cdvultur Can we get Example output here?

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
++
+[source,terminal]
+----
+$ oc logs -fn openshift-kmm deployments/kmm-operator-controller-manager
+----
++
+.Example output
+[source,terminal]
+----
+
```
Comment (StephenJamesSmith) [inline modules/kmm-gathering-data-for-kmm.adoc:38]: @cdvultur Can we get Example output here?

File: modules/kmm-gathering-data-for-kmm-hub.adoc (hunk 1 above)
Comment (StephenJamesSmith) [inline modules/kmm-gathering-data-for-kmm-hub.adoc:38]: @cdvultur Do we need to document collecting audit logs? Ex: $ oc adm must-gather -- /usr/bin/gather_audit_logs

File: modules/kmm-gathering-data-for-kmm.adoc (hunk 2 above)
Comment (cdvultur) [reply modules/kmm-gathering-data-for-kmm.adoc:38]: Sure, feel free to tune it as needed. 

As in previous example here are the first 2 line, ommited due to known bug:

```
E0228 09:36:36.191293       1 main.go:92] kmm/setup "msg"="Could not get the git commit; using <undefined>" "error"="vcs.revision not found in build info settings" 
I0228 09:36:36.191387       1 main.go:104] kmm/setup "msg"="Creating manager" "git commit"="<undefined>"
```


Here it is:

```
I0228 09:36:37.352405       1 request.go:682] Waited for 1.001998746s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/machine.openshift.io/v1beta1?timeout=32s
I0228 09:36:40.767060       1 listener.go:44] kmm/controller-runtime/metrics "msg"="Metrics server is starting to listen" "addr"="127.0.0.1:8080"
I0228 09:36:40.769483       1 main.go:234] kmm/setup "msg"="starting manager" 
I0228 09:36:40.769907       1 internal.go:366] kmm "msg"="Starting server" "addr"={"IP":"127.0.0.1","Port":8080,"Zone":""} "kind"="metrics" "path"="/metrics"
I0228 09:36:40.770025       1 internal.go:366] kmm "msg"="Starting server" "addr"={"IP":"::","Port":8081,"Zone":""} "kind"="health probe"
I0228 09:36:40.770128       1 leaderelection.go:248] attempting to acquire leader lease openshift-kmm/kmm.sigs.x-k8s.io...
I0228 09:36:40.784396       1 leaderelection.go:258] successfully acquired lease openshift-kmm/kmm.sigs.x-k8s.io
I0228 09:36:40.784876       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module" "source"="kind source: *v1beta1.Module"
I0228 09:36:40.784925       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module" "source"="kind source: *v1.DaemonSet"
I0228 09:36:40.784968       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module" "source"="kind source: *v1.Build"
I0228 09:36:40.785001       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module" "source"="kind source: *v1.Job"
I0228 09:36:40.785025       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module" "source"="kind source: *v1.Node"
I0228 09:36:40.785039       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="Module" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="Module"
I0228 09:36:40.785458       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="PodNodeModule" "controllerGroup"="" "controllerKind"="Pod" "source"="kind source: *v1.Pod"
I0228 09:36:40.786947       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="PreflightValidation" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidation" "source"="kind source: *v1beta1.PreflightValidation"
I0228 09:36:40.787406       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="PreflightValidation" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidation" "source"="kind source: *v1.Build"
I0228 09:36:40.787474       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="PreflightValidation" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidation" "source"="kind source: *v1.Job"
I0228 09:36:40.787488       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="PreflightValidation" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidation" "source"="kind source: *v1beta1.Module"
I0228 09:36:40.787603       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="NodeKernel" "controllerGroup"="" "controllerKind"="Node" "source"="kind source: *v1.Node"
I0228 09:36:40.787634       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="NodeKernel" "controllerGroup"="" "controllerKind"="Node"
I0228 09:36:40.787680       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="PreflightValidation" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidation"
I0228 09:36:40.785607       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream" "source"="kind source: *v1.ImageStream"
I0228 09:36:40.787822       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="preflightvalidationocp" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidationOCP" "source"="kind source: *v1beta1.PreflightValidationOCP"
I0228 09:36:40.787853       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream"
I0228 09:36:40.787879       1 controller.go:185] kmm "msg"="Starting EventSource" "controller"="preflightvalidationocp" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidationOCP" "source"="kind source: *v1beta1.PreflightValidation"
I0228 09:36:40.787905       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="preflightvalidationocp" "controllerGroup"="kmm.sigs.x-k8s.io" "controllerKind"="PreflightValidationOCP"
I0228 09:36:40.786489       1 controller.go:193] kmm "msg"="Starting Controller" "controller"="PodNodeModule" "controllerGroup"="" "controllerKind"="Pod"
```

File: modules/kmm-gathering-data-for-kmm-hub.adoc (hunk 1 above)
Comment (cdvultur) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: Not sure on the `gather_audit_logs` . Tried to run it on my setup and I get `/usr/bin/gather_audit_logs: No such file or directory`

Maybe @qbarrand can help on this one

```
$ oc adm must-gather --image="${MUST_GATHER_IMAGE}" -- /usr/bin/gather_audit_logs
[must-gather      ] OUT Using must-gather plug-in image: registry.redhat.io/kmm/kernel-module-management-must-gather-rhel8@sha256:4a6b25ba3b4302e940ebbdacfeb6280a008edd56475363cd93087dcfc322d344
When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
ClusterID: 6ad73595-913b-4fae-a52e-ac70b3c98ae0
ClusterVersion: Stable at "4.12.0-0.nightly-2023-02-22-173620"
ClusterOperators:
	All healthy and stable


[must-gather      ] OUT namespace/openshift-must-gather-57g79 created
[must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-4th8k created
[must-gather      ] OUT pod for plug-in image registry.redhat.io/kmm/kernel-module-management-must-gather-rhel8@sha256:4a6b25ba3b4302e940ebbdacfeb6280a008edd56475363cd93087dcfc322d344 created
[must-gather-x6tg7] POD 2023-04-17T11:27:22.226561557Z /bin/bash: /usr/bin/gather_audit_logs: No such file or directory
[must-gather-x6tg7] OUT waiting for gather to complete

```

Comment (cdvultur) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: Sure, here it is:

Note that for both kmm and kmm-hub operators, I have omitted from output the first two lines, 

```
E0417 11:34:07.754997       1 main.go:94] kmm-hub/setup "msg"="Could not get the git commit; using <undefined>" "error"="vcs.revision not found in build info settings" 
I0417 11:34:07.755111       1 main.go:98] kmm-hub/setup "msg"="Creating manager" "git commit"="<undefined>"
```
which is a bug. Once that is fixed it should specify the good version from which it was built. 
Let me know if we should keep it in there or not. 


```
$ oc logs -fn openshift-kmm-hub deployments/kmm-operator-hub-controller-manager
I0417 11:34:08.807472       1 request.go:682] Waited for 1.023403273s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/tuned.openshift.io/v1?timeout=32s
I0417 11:34:12.373413       1 listener.go:44] kmm-hub/controller-runtime/metrics "msg"="Metrics server is starting to listen" "addr"="127.0.0.1:8080"
I0417 11:34:12.376253       1 main.go:150] kmm-hub/setup "msg"="Adding controller" "name"="ManagedClusterModule"
I0417 11:34:12.376621       1 main.go:186] kmm-hub/setup "msg"="starting manager" 
I0417 11:34:12.377690       1 leaderelection.go:248] attempting to acquire leader lease openshift-kmm-hub/kmm-hub.sigs.x-k8s.io...
I0417 11:34:12.378078       1 internal.go:366] kmm-hub "msg"="Starting server" "addr"={"IP":"127.0.0.1","Port":8080,"Zone":""} "kind"="metrics" "path"="/metrics"
I0417 11:34:12.378222       1 internal.go:366] kmm-hub "msg"="Starting server" "addr"={"IP":"::","Port":8081,"Zone":""} "kind"="health probe"
I0417 11:34:12.395703       1 leaderelection.go:258] successfully acquired lease openshift-kmm-hub/kmm-hub.sigs.x-k8s.io
I0417 11:34:12.396334       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "source"="kind source: *v1beta1.ManagedClusterModule"
I0417 11:34:12.396403       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "source"="kind source: *v1.ManifestWork"
I0417 11:34:12.396430       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "source"="kind source: *v1.Build"
I0417 11:34:12.396469       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "source"="kind source: *v1.Job"
I0417 11:34:12.396522       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "source"="kind source: *v1.ManagedCluster"
I0417 11:34:12.396543       1 controller.go:193] kmm-hub "msg"="Starting Controller" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule"
I0417 11:34:12.397175       1 controller.go:185] kmm-hub "msg"="Starting EventSource" "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream" "source"="kind source: *v1.ImageStream"
I0417 11:34:12.397221       1 controller.go:193] kmm-hub "msg"="Starting Controller" "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream"
I0417 11:34:12.498335       1 filter.go:196] kmm-hub "msg"="Listing all ManagedClusterModules" "managedcluster"="local-cluster"
I0417 11:34:12.498570       1 filter.go:205] kmm-hub "msg"="Listed ManagedClusterModules" "count"=0 "managedcluster"="local-cluster"
I0417 11:34:12.498629       1 filter.go:238] kmm-hub "msg"="Adding reconciliation requests" "count"=0 "managedcluster"="local-cluster"
I0417 11:34:12.498687       1 filter.go:196] kmm-hub "msg"="Listing all ManagedClusterModules" "managedcluster"="sno1-0"
I0417 11:34:12.498750       1 filter.go:205] kmm-hub "msg"="Listed ManagedClusterModules" "count"=0 "managedcluster"="sno1-0"
I0417 11:34:12.498801       1 filter.go:238] kmm-hub "msg"="Adding reconciliation requests" "count"=0 "managedcluster"="sno1-0"
I0417 11:34:12.501947       1 controller.go:227] kmm-hub "msg"="Starting workers" "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream" "worker count"=1
I0417 11:34:12.501948       1 controller.go:227] kmm-hub "msg"="Starting workers" "controller"="ManagedClusterModule" "controllerGroup"="hub.kmm.sigs.x-k8s.io" "controllerKind"="ManagedClusterModule" "worker count"=1
I0417 11:34:12.502285       1 imagestream_reconciler.go:50] kmm-hub "msg"="registered imagestream info mapping" "ImageStream"={"name":"driver-toolkit","namespace":"openshift"} "controller"="imagestream" "controllerGroup"="image.openshift.io" "controllerKind"="ImageStream" "dtkImage"="quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:df42b4785a7a662b30da53bdb0d206120cf4d24b45674227b16051ba4b7c3934" "name"="driver-toolkit" "namespace"="openshift" "osImageVersion"="412.86.202302211547-0" "reconcileID"="e709ff0a-5664-4007-8270-49b5dff8bae9"
```

Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: @cdvultur Thanks for this. Do you have an estimate when that bug will be fixed? Is it appropriate to show the output without those 2 lines?

Comment (cdvultur) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: Not sure on this. The issue is in progress. 

@qbarrand do you have an eta on the git version issue. Should we add it in the docs ?

Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: @cdvultur @qbarrand - Added the example output minus the first two lines. Please let me know if this is acceptable to publish.

File: modules/kmm-gathering-data-for-kmm.adoc (hunk 2 above)
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc]: @cdvultur @qbarrand - Added the example output minus the first two lines. Please let me know if this is acceptable to publish.

**Comment by StephenJamesSmith** [comment]
@qbarrand @cdvultur If you are finished with your reviews and are satisfied with the content, please add the comment "/LGTM" so I can push this to Peer Review. Thx!

File: modules/kmm-gathering-data-for-kmm-hub.adoc (hunk 1 above)
Comment (qbarrand) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:38]: Our must-gather image should collect everything we need. Thank you for bringing this up though.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Sure, here it is:
+
+Note that for both kmm and kmm-hub operators, I have omitted from output the first two lines,
+
+E0417 11:34:07.754997       1 main.go:94] kmm-hub/setup "msg"="Could not get the git commit; using <undefined>" "error"="vcs.revision not found in build info settings"
+I0417 11:34:07.755111       1 main.go:98] kmm-hub/setup "msg"="Creating manager" "git commit"="<undefined>"
+
+which is a bug. Once that is fixed it should specify the good version from which it was built.
+Let me know if we should keep it in there or not.
+
```
Comment (qbarrand) [inline modules/kmm-gathering-data-for-kmm-hub.adoc:49]: I don't think this should be included in the docs.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
++
+[source,terminal]
+----
+export MUST_GATHER_IMAGE=$(oc get deployment -n openshift-kmm-hub kmm-operator-hub-controller-manager -ojsonpath='{.spec.template.spec.containers[?(@.name=="manager")].env[?(@.name=="RELATED_IMAGES_MUST_GATHER")].value}')
+----
++
+[source,terminal]
+----
+$ oc adm must-gather --image="${MUST_GATHER_IMAGE}" -- /usr/bin/gather -u
+----
```
Comment (adellape) [inline modules/kmm-gathering-data-for-kmm-hub.adoc:21]: Ideally, these two commands (here and in the equivalent KMM Operator proc's module) should be separate steps or substeps. Maybe like:

```
. Gather the data for the KMM-Hub Operator controller manager:

.. Set the `MUST_GATHER_IMAGE` variable:
+
[source,terminal]
----
$ export MUST_GATHER_IMAGE= [...]
----

.. Run the `must-gather` tool:
+ 
[source,terminal]
----
$ oc adm must-gather [...]
----
```

File: hardware_enablement/kmm-kernel-module-management.adoc [hunk 5]
```diff
@@ -80,3 +80,13 @@ include::modules/kmm-configuring-the-lookup-path-on-nodes.adoc[leveloffset=+2]
 
 include::modules/kmm-building-a-moduleloader-image.adoc[leveloffset=+2]
 include::modules/kmm-tuning-the-module-resource.adoc[leveloffset=+2]
+
+// Added for TELCODOCS-1059
+include::modules/kmm-troubleshooting.adoc[leveloffset=+1]
+include::modules/kmm-must-gather-tool.adoc[leveloffset=+2]
+[role="_additional-resources"]
+.Additional resources
+xref:../serverless/serverless-support.adoc#about-must-gather[About the must-gather tool]
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply hardware_enablement/kmm-kernel-module-management.adoc:89]: Added.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+[id="kmm-gathering-data-for-kmm-hub_{context}"]
+= Gathering data for KMM-Hub
+
+.Procedure
+
+. Gather the data for the KMM operator hub controller manager:
++
+[source,terminal]
+----
+export MUST_GATHER_IMAGE=$(oc get deployment -n openshift-kmm-hub kmm-operator-hub-controller-manager -ojsonpath='{.spec.template.spec.containers[?(@.name=="manager")].env[?(@.name=="RELATED_IMAGES_MUST_GATHER")].value}')
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:15]: Added "$".

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+----
++
+[source,terminal]
+----
+$ oc adm must-gather --image="${MUST_GATHER_IMAGE}" -- /usr/bin/gather -u
+----
++
+[NOTE]
+====
+Use the `-n NAMESPACE` switch to specify a namespace if you installed KMM in a custom namespace.
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:25]: Docs in the upstream repo describe this as a switch.  Changed to <namespace>.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+====
+
+. View the operator logs:
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:28]: changed.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
++
+[source,terminal]
+----
+$ oc logs -fn openshift-kmm-hub deployments/kmm-operator-hub-controller-manager
+----
++
+.Example output
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:35]: Added this option. Thx!

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-gathering-data-for-kmm_{context}"]
+= Gathering data for KMM
+
+.Procedure
+
+. Gather the data for the KMM operator controller manager:
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc:11]: Changed.

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
++
+[source,terminal]
+----
+export MUST_GATHER_IMAGE=$(oc get deployment -n openshift-kmm kmm-operator-controller-manager -ojsonpath='{.spec.template.spec.containers[?(@.name=="manager")].env[?(@.name=="RELATED_IMAGES_MUST_GATHER")].value}')
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc:15]: Added "$".

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+----
++
+[source,terminal]
+----
+$ oc adm must-gather --image="${MUST_GATHER_IMAGE}" -- /usr/bin/gather
+----
++
+[NOTE]
+====
+Use the `-n NAMESPACE` switch to specify a namespace if you installed KMM in a custom namespace.
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc:25]: Changed "NAMESPACE". Left "switch" - see previous reply.

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+====
+
+. View the operator logs:
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc:28]: Changed.

File: modules/kmm-gathering-data-for-kmm.adoc [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
++
+[source,terminal]
+----
+$ oc logs -fn openshift-kmm deployments/kmm-operator-controller-manager
+----
++
+.Example output
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm.adoc:35]: Added.

File: modules/kmm-gathering-data-for-kmm-hub.adoc [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-gathering-data-for-kmm-hub_{context}"]
+= Gathering data for KMM-Hub
+
+.Procedure
+
+. Gather the data for the KMM operator hub controller manager:
```
[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-gathering-data-for-kmm-hub.adoc:11]: All good questions. "KMM Operator hub controller manager" seems right and consistent with the other module and the code. 

"KMM-Hub" is how it is used in the upstream repo.

**Pull request merged**